File: Main/Old NX/Document.vb
Public Class Document
    Public Document As String
    Public Path As String
    Public Title As String

    Sub Save(ByVal overrideFileNAme As Boolean, Optional ByVal newFileNAme As String = "")
        If Not overrideFileNAme Then
            IO.File.WriteAllText(Path, Document)
        Else
            IO.File.WriteAllText(newFileNAme, Document)
        End If
    End Sub

    Sub New(ByVal title As String, ByVal path As String, ByVal document As String)
        Me.Document = document
        Me.Path = path
        Me.Title = title
    End Sub
End Class

File: Plugins/CodeEditor/Document.cs
using Microsoft.VisualBasic;
using System;
using System.Collections;
using System.Collections.Generic;
using System.Data;
using System.Diagnostics;
using System.Linq;
using System.Xml.Linq;
namespace CodeEditor
{
	public class Document
	{
		public string DocumentText;
		public string Path;

		public string Title;
		public void Save(bool overrideFileNAme, string newFileNAme = "")
		{
			if (!overrideFileNAme) {
				System.IO.File.WriteAllText(Path, DocumentText);
			} else {
				System.IO.File.WriteAllText(newFileNAme, DocumentText);
			}
		}

		public Document(string title, string path, string document)
		{
			this.DocumentText = document;
			this.Path = path;
			this.Title = title;
		}
	}
}
